src/Core/Content/Media/Aggregate/MediaThumbnail/MediaThumbnailEntity.php

```diff
@@ -22,10 +23,24 @@ class MediaThumbnailEntity extends Entity
 
     protected ?string $url = '';
 
-    protected string $mediaId;
+    /**
+     * @deprecated tag:v6.8.0 - Will be non-nullable
+     */
+    protected ?string $mediaId;
 
     protected ?MediaEntity $media = null;
 
+    public function assign(array $options)
+    {
+        parent::assign($options);
+
+        if (!isset($this->mediaId) && Feature::isActive('v6.8.0.0')) {
```

Comment:
the feature flag check should be removed here, otherwise there will be no deprecation warning within 6.7

```suggestion
        if (!isset($this->mediaId)) {
```